python/bindings_manif.cpp

```diff
@@ -10,7 +10,7 @@ void wrap_SE3(pybind11::module &m);
 
 void wrap_SE_2_3(pybind11::module &m);
 
-PYBIND11_MODULE(manifpy, m) {
+PYBIND11_MODULE(_bindings, m) {
```

Comment:
Genuine question, why the renaming? 
Altho unlikely, `_binding` may be a little too generic and cause conflicts. How about `_manifpy`, `_manif_bindings` or such?

Reply:
This is related to the first point of https://github.com/artivis/manif/pull/233#issuecomment-862604986.

Referring to the installed tree in the active site-packages, now the bindings file is no longer the main entry point of the package. In fact, now a new `manifpy` folder is created and the bindings file (the cpython shared lib) is installed inside it.

Before this PR, manifpy could be used as:

```python
import manifpy
from manifpy import SE3

se2 = manifpy.SE2()
se3 = SE3()
```

This PR is backward compatible with that usage because the `__init__.py` file contains `from _bindings import *`. This also means that dowstream users, in addition to the snippet avobe, can also do the following:

```
import manifpy._bindings

se2 = manifpy._bindings.SE2()
se3 = manifpy._bindings.SE3()
```

The leading `_` in python usually means a private item, and while the last snippet is totally legit, it implies that the real C module is an implementation detail. It can have any name you like, I don't have strong opinions on it.

In light of the explanation above, please let me know the name you prefer. I could even revert to the original name that would result in the following valid snippet:

```
import manifpy
from manifpy import SE3
import manifpy.manifpy

se2 = manifpy.SE2()
se3 = SE3()
so3 = manifpy.manifpy.SO3()
```

I like using `bindings` or `_bindings` for the real cpython module, but it's just personal taste and there is no common pattern I noticed around.

<details>
<summary>Package Tree</summary>

```
manif on  feature/setuptools via △ v3.20.2 🅒 /conda  
❯ tree /conda/lib/python3.8/site-packages/manifpy
/conda/lib/python3.8/site-packages/manifpy
├── __init__.py
├── _bindings.cpython-38-x86_64-linux-gnu.so
├── include
│   ├── manif
│   │   ├── Rn.h
│   │   ├── SE2.h
│   │   ├── SE3.h
│   │   ├── SE_2_3.h
│   │   ├── SO2.h
│   │   ├── SO3.h
│   │   ├── algorithms
│   │   │   ├── average.h
│   │   │   ├── bezier.h
│   │   │   ├── decasteljau.h
│   │   │   └── interpolation.h
│   │   ├── ceres
│   │   │   ├── ceres.h
│   │   │   ├── ceres_utils.h
│   │   │   ├── constants.h
│   │   │   ├── constraint.h
│   │   │   ├── local_parametrization.h
│   │   │   └── objective.h
│   │   ├── constants.h
│   │   ├── functions.h
│   │   ├── gtest
│   │   │   ├── gtest_eigen_utils.h
│   │   │   └── gtest_manif_utils.h
│   │   ├── impl
│   │   │   ├── assignment_assert.h
│   │   │   ├── eigen.h
│   │   │   ├── generator.h
│   │   │   ├── lie_group_base.h
│   │   │   ├── macro.h
│   │   │   ├── random.h
│   │   │   ├── rn
│   │   │   │   ├── Rn.h
│   │   │   │   ├── RnTangent.h
│   │   │   │   ├── RnTangent_base.h
│   │   │   │   ├── RnTangent_map.h
│   │   │   │   ├── Rn_base.h
│   │   │   │   ├── Rn_map.h
│   │   │   │   └── Rn_properties.h
│   │   │   ├── se2
│   │   │   │   ├── SE2.h
│   │   │   │   ├── SE2Tangent.h
│   │   │   │   ├── SE2Tangent_base.h
│   │   │   │   ├── SE2Tangent_map.h
│   │   │   │   ├── SE2_base.h
│   │   │   │   ├── SE2_map.h
│   │   │   │   └── SE2_properties.h
│   │   │   ├── se3
│   │   │   │   ├── SE3.h
│   │   │   │   ├── SE3Tangent.h
│   │   │   │   ├── SE3Tangent_base.h
│   │   │   │   ├── SE3Tangent_map.h
│   │   │   │   ├── SE3_base.h
│   │   │   │   ├── SE3_map.h
│   │   │   │   └── SE3_properties.h
│   │   │   ├── se_2_3
│   │   │   │   ├── SE_2_3.h
│   │   │   │   ├── SE_2_3Tangent.h
│   │   │   │   ├── SE_2_3Tangent_base.h
│   │   │   │   ├── SE_2_3Tangent_map.h
│   │   │   │   ├── SE_2_3_base.h
│   │   │   │   ├── SE_2_3_map.h
│   │   │   │   └── SE_2_3_properties.h
│   │   │   ├── so2
│   │   │   │   ├── SO2.h
│   │   │   │   ├── SO2Tangent.h
│   │   │   │   ├── SO2Tangent_base.h
│   │   │   │   ├── SO2Tangent_map.h
│   │   │   │   ├── SO2_base.h
│   │   │   │   ├── SO2_map.h
│   │   │   │   └── SO2_properties.h
│   │   │   ├── so3
│   │   │   │   ├── SO3.h
│   │   │   │   ├── SO3Tangent.h
│   │   │   │   ├── SO3Tangent_base.h
│   │   │   │   ├── SO3Tangent_map.h
│   │   │   │   ├── SO3_base.h
│   │   │   │   ├── SO3_map.h
│   │   │   │   └── SO3_properties.h
│   │   │   ├── tangent_base.h
│   │   │   ├── traits.h
│   │   │   └── utils.h
│   │   └── manif.h
│   └── tl
│       └── optional.hpp
└── share
    └── manif
        └── cmake
            ├── manifConfig.cmake
            ├── manifConfigVersion.cmake
            └── manifTargets.cmake
```

</details>

Reply:
Thanks for the detailed explanation. Let's leave it as it is, I guess there is no actual chance for a naming conflict.